### PR TITLE
[SYCL] Add tests for some half builtins

### PR DESCRIPTION
**Pull request by t4c1**
Added tests for some libclc builtins that work on halfs. We will be making some changes to those, but the tests for them can be merged ahead of time.

Tests some of the changes in https://github.com/intel/llvm/pull/5724

### CODE REVIEW

**Comment by t4c1** [comment]
@bader Can I ask you for a copy of the logs for failed CI tests?

**Comment by bader** [comment]
To run this test on OpenCL back-end, the target must support cl_khr_fp16 extension. Intel OpenCL CPU doesn't support it.

**Comment by t4c1** [comment]
Is it possible to specify an extension in XFAIL? Or should I just set XFAIL for OpenCL backend?

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp:3]: There are 3 ways to fix the pre-commit:
1. Remove execution on CPU
2. Mark as expected to fail on CPU or OpenCL CPU -> `XFAIL: cpu` or `XFAIL: cpu && opencl`
3. Add "UNSUPPORTED: cpu".

I prefer 3 as it saves the time to run the test.

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]: Please, modernize SYCL API i.e. include <sycl/sycl.hpp>, replace `cl::sycl` namespace to `sycl`.

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]:
```suggestion
             std::numeric_limits<half>::epsilon() ||
         a < std::numeric_limits<half>::min();

```

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]:
```suggestion
#include <limits>
```
This test doesn't use `std::unordered_set`.

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]:
```suggestion
constexpr int N = 16; // divisible by all vector sizes
```
Looking that macro processing 3-element vector I see that N is not required to be divisible by 3.

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]: I think it would be more efficient to have a kernel testing all functions instead of a kernel per function.

I don't know how to apply the same logic for different vector length (i.e. reduce # of test kernels), but we probably can consider allocating more buffers (e.g. one buffer per vector length).

File: SYCL/Basic/half_builtins.cpp
Comment (bader) [inline SYCL/Basic/half_builtins.cpp]:
```suggestion
#include <sycl/sycl.hpp>

```

**Review by bader — DISMISSED** [review]

**Comment by t4c1** [comment]
Is this PR waiting for anything?

**Comment by bader** [comment]
Waiting on code owner review from intel/llvm-reviewers-runtime.
@intel/llvm-reviewers-runtime, @smaslov-intel, ping.

File: SYCL/Basic/half_builtins.cpp
Comment (smaslov-intel) [inline SYCL/Basic/half_builtins.cpp:8]: NIT: would it be better to test for the extension support in the source code itself?

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Pull request merged**
